lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -170,16 +171,20 @@ def _on_leader_elected(self, event: LeaderElectedEvent):
 
         if not self.peers_data.get(Scope.APP, "admin_user_initialized"):
             self.unit.status = MaintenanceStatus(AdminUserInitProgress)
-            # User config is currently in a default state, which contains multiple insecure default
-            # users. Purge the user list before initialising the users the charm requires.
-            self._purge_users()
-            self._put_admin_user()
-            self.peers_data.put(Scope.APP, "admin_user_initialized", True)
-            self.status.clear(AdminUserInitProgress)
+
+        # User config is currently in a default state, which contains multiple insecure default
+        # users. Purge the user list before initialising the users the charm requires.
+        self._purge_users()
```

Comment:
What happens if we delete the leader node in a cluster that's already running?

Reply:
It should not be a problem as long as the security index is initialized, as this file is only loaded once and besides the admin user (which is added after this step) all other users are added through the rest api.
In other words, the only unit(s) where this file is touched (with or without effect) are leader units, if the cluster is already initialized => no harm done. If not initialized => expected